core/src/main/java/org/elasticsearch/index/translog/TranslogReader.java

```diff
@@ -115,8 +117,9 @@ public static TranslogReader open(
                     case TranslogWriter.VERSION_CHECKSUMS:
                         throw new IllegalStateException("pre-2.0 translog found [" + path + "]");
                     case TranslogWriter.VERSION_CHECKPOINTS:
+                    case TranslogWriter.VERSION_TIMESTAMPS:
                         assert path.getFileName().toString().endsWith(Translog.TRANSLOG_FILE_SUFFIX) : "new file ends with old suffix: " + path;
-                        assert checkpoint.numOps >= 0 : "expected at least 0 operatin but got: " + checkpoint.numOps;
+                        assert checkpoint.numOps >= 0 : "expected at least 0 operation but got: " + checkpoint.numOps;
```

Comment:
tja, maybe. I tried to avoid the entire what metadata is supported where discussion. I mean we walk to great length to avoid directory traversals. I think this is simple, solid and entirely self contained?